pkg/network/protocols/http/protocol.go

```diff
@@ -216,13 +216,13 @@ func (p *protocol) setupMapCleaner(mgr *manager.Manager) {
 
 // GetStats returns a map of HTTP stats stored in the following format:
 // [source, dest tuple, request path] -> RequestStats object
-func (p *protocol) GetStats() *protocols.ProtocolStats {
+func (p *protocol) GetStats() (*protocols.ProtocolStats, func()) {
```

Comment:
Maybe we can improve the function name as now it is not returning only the stats

Reply:
Do you have a suggestion?
The function does return the stats, and the cleaner is not something that should be mentioned in the function name

Reply:
Maybe something like `GetStatsAndCleanup`, but it's not the best name. I'll leave it up to you if you want to change it or not

Reply:
Deferred cleanup is a common pattern in go, I don't think it provides any additional information while there's a good documentation